src/utils/fhirFilter.ts

```diff
@@ -93,53 +100,131 @@ export const addCancerHistologyMorphology = (
   return condition;
 };
 
-export function convertStringtoResource({
-  bundle,
+export function convertStringToObservation({
   valueString,
   id,
   profile_value,
   codingSystem,
   codingSystemCode,
 }: {
-  bundle: Bundle;
   valueString: string | string[];
   id: string;
   profile_value: string;
   codingSystem: string;
   codingSystemCode: string;
-}): void {
+}): Observation {
+  // Create the Condition - done 1separate from the function call to ensure proper TypeScript checking
+  let retResource: Observation = null;
+  if (valueString != null) {
+    let code: CodeableConcept = null;
+    if (codingSystemCode) {
+      code = {
+        coding: [
+          {
+            system: codingSystem,
+            code: codingSystemCode,
+          },
+        ],
+      };
+    }
+
+    if (code) {
+      const resource: Observation = {
+        resourceType: 'Observation',
+        id: id,
+        meta: {
+          profile: [profile_value],
+        },
+        code,
+        valueString,
+      };
+      retResource = resource;
+    } else {
+      const resource: Observation = {
+        resourceType: 'Observation',
+        id: id,
+        meta: {
+          profile: [profile_value],
+        },
+        valueString,
+      };
+      retResource = resource;
+    }
+  }
+  return retResource;
+}
+export function convertCodedValueToObervation({
+  codedValue,
+  id,
+  profile_value,
+  codingSystem,
+}: {
+  codedValue: CodedValueType;
+  id: string;
+  profile_value: string;
+  codingSystem: string;
+}): Observation {
+  // Create the Condition - done separate from the function call to ensure proper TypeScript checking
+
+  const tmpCode: string | number = codedValue.code.toString();
+  const tmpDisplay = codedValue.display;
+
+  const resource: Observation = {
+    resourceType: 'Observation',
+    id: id,
+    status: 'completed',
+    subject: search_patient,
```

Comment:
The subject information should probably be generated elsewhere and passed in as a variable. It shouldn't be a constant value.